apps/docs/pages/guides/database/postgres/indexes.mdx

```diff
@@ -0,0 +1,107 @@
+
+Indexes can be structured in a few different ways. The type of index chosen often depends on the values you are indexing. By far the most common index type, and the default in Postgres, is the B-Tree. A B-Tree is the generalized form of a binary search tree, where nodes can have more than two children.
+
+Whilst indexes are useful for speeding up query performance, the query planner may not always make use of a given index when choosing which optimizations to make. Additionally indexes come with some overhead in terms of additional writes and storage, so it's useful to understand how and when to use indexes, if at all.
+
+## Create an index
+
+Let's take an example table:
+```sql
+create table persons (
+  id bigint generated by default as identity primary key,
+  age int,
+  height int,
+  weight int,
+  name text,
+  deceased boolean
+);
+```
+<Admonition>
+
+All the queries in this guide can be run using the [SQL Editor](https://app.supabase.com/project/_/sql) in the Supabase Dashboard, or via `psql` if you're [connecting directly to the database](/docs/guides/database/connecting-to-postgres#direct-connections).
+
+</Admonition>
+
+We might want to frequently query users based on their age:
+```sql
+select name from persons where age = 32;
+```
+
+The issue is that without an index, postgres will do a sequential scan of every row in the table to find equality matches on age.
+
+You can verify this by doing an explain on the query:
+```sql
+explain select name from persons where age = 32;
+```
+
+Outputs:
+```
+Seq Scan on persons  (cost=0.00..22.75 rows=x width=y)
+Filter: (age = 32)
+```
+
+To add a simple B-Tree index you can run:
+```sql
+create index idx_persons_age on persons(age);
+```
+
+<Admonition type="caution">
+
+It can take a long time to build indexes on large datasets and the default behaviour of `create index` is to lock the table from writes.
+
+Luckily Postgres provides us with `create index concurrently` which prevents blocking writes on the table, but does take a bit longer to build.
+
+</Admonition>
+
+Here is a simplified diagram of the index we just created (note that in practice, nodes actually have more than two children).
+
+![B-Tree index example in Postgres](/docs/img/b-tree-example.png)
+
+You can see that in any large data set, traversing the index to locate a given value can be done in much less operations (O(log n)) than compared to scanning the table one value at a time from top to bottom (O(n)).
+
+## Partial indexes
+
+If you are frequently querying only a subset of rows then it may be more performant and space efficient to build a partial index. In our example, perhaps we only want to match on `age` where `deceased is false`. We could build a partial index:
```

Comment:
```suggestion
If you are frequently querying a subset of rows then it may be more efficient to build a partial index. In our example, perhaps we only want to match on `age` where `deceased is false`. We could build a partial index:
```